rclcpp/include/rclcpp/executor.hpp

```diff
@@ -319,6 +320,53 @@ class Executor
   virtual void
   spin_once(std::chrono::nanoseconds timeout = std::chrono::nanoseconds(-1));
 
+  // TODO(hliberacki): Add documentation
+  template<typename FutureT, typename DurationT = std::chrono::milliseconds,
+    // typename std::enable_if_t<
```

Comment:
I believe Humble targets C++17, so I see no reason why not to use any C++17 features you like, but I am definitely not the right person to ask. Someone at OR would have to jump in to let us know if we should be trying to minimize its use to help in C++14 compatibility for other important users

Reply:
> typename std::enable_if_t<!std::is_invocable_v<FutureT>, bool> = true

It'll be hard to understand in the future why `std::is_invocable_v` was used.
I would rather have the more accurate condition (even if it doesn't look pretty) or two overloads.

Reply:
@ivanpauno At the moment i've decided to use `if constexpr` and reduce second check whether the `Condition` is a of type `Future` or `SharedFuture`. I did it because realize that there are more use cases, like using `src/ros2/rclcpp/rclcpp/include/rclcpp/client.hpp` from`src/ros2/rclcpp/rclcpp/include/rclcpp/client.hpp`. 

If you still would like to  have more grain granularity - then I would suggest to create Concept like requirement of the API for the given condition. So it would have to have `wait_for(timeout)` method which returns `enum class future_status`. I did not introduce it for now, on purpose because I am not sure whether it will be valuable to have that type_trait which is a bit hard to read. But If needed I can introduce it - no problem with that.